File: src/jskills/factorgraphs/Variable.java
package jskills.factorgraphs;

import lombok.Getter;
import lombok.Setter;

public class Variable<TValue>
{
    private final String name;
    private final TValue prior;
    @Getter @Setter private TValue value;

    public Variable(TValue prior, String name, Object... args) {
        this.name = "Variable[" + String.format(name, args) + "]";
        this.prior = prior;
        resetToPrior();
    }

    public void resetToPrior() { value = prior; }

    @Override public String toString() { return name; }
}

File: TrueSkillEngine/src/main/scala/jskills/factorgraphs/Variable.scala
package jskills.factorgraphs

class Variable[T](prior: T, name: String) {
  resetToPrior()

  var value: T = prior

  def resetToPrior() { value = prior }

  override def toString() = name
}

object Variable {
  def apply[T](prior: T, name: String, args: Any*) = {
    new Variable[T](prior, String.format(name, args))
  }
}